src/Processors/Formats/Impl/DiagramOutputFormat.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include <cstddef>
+#include <vector>
+#include <Formats/FormatFactory.h>
+#include <Formats/FormatSettings.h>
+#include <Processors/Formats/IOutputFormat.h>
+
+namespace DB
+{
+
+
+class DiagramOutputFormat : public IOutputFormat
+{
+    enum class DiagramType
+    {
+        SCATTER,
+        LINEPLOT,
+        HISTOGRAM,
+        HEAT_MAP,
+    };
+    struct Color
+    {
+        Float64 r = 0;
+        Float64 g = 0;
+        Float64 b = 0;
+    };
+    struct BraileSymbol
+    {
+        ssize_t points[4][2] = {{-1, -1}, {-1, -1}, {-1, -1}, {-1, -1}};
+        ssize_t getColor() const;
+        UInt8 getSymbolNum() const;
+    };
+    static inline const std::unordered_map<String, DiagramType> diagram_type
+        = {{"SCATTER", DiagramType::SCATTER},
+           {"HISTOGRAM", DiagramType::HISTOGRAM},
+           {"LINEPLOT", DiagramType::LINEPLOT},
```

Comment:
It's inconsistent, because others are also plots. Should be just "line".
Let's make all lowercase for simplicity. Let's name heatmap as "heatmap".